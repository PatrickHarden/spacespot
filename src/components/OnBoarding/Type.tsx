import React from 'react'
import { useIntl } from 'react-intl'
import styled from 'styled-components'
import OutlineButton, { ButtonProps } from './OutlineButton'
import Heading4 from 'components/common/Heading4'

const Container = styled.div`
  padding: 0%;
  margin-bottom: 50px;
`
const ButtonContainer = styled.div`
  display: flex;
  max-width: 360px;
  flex: 1;
`
const TypeTitle = styled.div`
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
`
const TypeSubTitle = styled.div`
  color: #828286;
  font-size: 14px;
  font-weight: 400;
`
const TypeButton = styled(OutlineButton)<ButtonProps & { isLeft?: boolean }>`
  text-align: center;
  width: 47%
  padding: 28px 0;
  margin-right: ${props => (props.isLeft ? '6%' : '0')};
  :hover { ${props =>
    props.selected
      ? ''
      : `border: 1px solid #6cb9d5;
         box-sizing: border-box;
         box-shadow: 0 1px 4px 0 #6cb9d5;
         cursor: pointer`}
`

export interface TypeProps {
  onClick: (type: string) => void
  selected?: string
}

const Type = (props: TypeProps) => {
  const { onClick, selected } = props
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })

  return (
    <Container>
      <Heading4>{t('ONBOARDING_TYPE_LEASE')}</Heading4>
      <ButtonContainer>
        <TypeButton
          testId="add-fixed"
          isLeft={true}
          onClick={() => onClick('FIXED')}
          selected={selected === 'FIXED'}>
          <TypeTitle>{t('ONBOARDING_TYPE_FIXED_STANDARD_OFFICE')}</TypeTitle>
          <TypeSubTitle>{t('ONBOARDING_TYPE_FIXED_SUBTITLE')}</TypeSubTitle>
        </TypeButton>
        <TypeButton
          testId="add-flex"
          onClick={() => onClick('FLEX')}
          selected={selected === 'FLEX'}>
          <TypeTitle>{t('ONBOARDING_TYPE_FLEX_COWORKING')}</TypeTitle>
          <TypeSubTitle>{t('ONBOARDING_TYPE_FLEX_SUBTITLE')}</TypeSubTitle>
        </TypeButton>
      </ButtonContainer>
    </Container>
  )
}
export default Type
