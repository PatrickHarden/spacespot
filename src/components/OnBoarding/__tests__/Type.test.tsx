import React from 'react'
import Type from '../Type'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

it('renders without crashing ', async () => {
  const setType = jest.fn()
  const wrapper = mount(<Type selected={'test'} onClick={setType} />)
  expect(mountToJson(wrapper)).toMatchSnapshot()
  const buttons = wrapper.find('button')
  act(() => {
    buttons.at(0).simulate('click')
  })
  expect(setType).toHaveBeenCalledWith('FIXED')
  act(() => {
    buttons.at(1).simulate('click')
  })
  expect(setType).toHaveBeenCalledWith('FLEX')
})
