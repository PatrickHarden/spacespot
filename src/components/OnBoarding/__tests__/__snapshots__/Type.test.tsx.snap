// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<Type
  onClick={[MockFunction]}
  selected="test"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "lastClassName": "heZAUk",
            "rules": Array [
              "
  padding: 0%;
  margin-bottom: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bxivhb heZAUk"
      >
        <styled.h4>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "ezOBiL",
                  "rules": Array [
                    "
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
",
                  ],
                },
                "displayName": "styled.h4",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "h4",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h4
              className="sc-htpNat ezOBiL"
            >
              ONBOARDING_TYPE_LEASE
            </h4>
          </StyledComponent>
        </styled.h4>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "fbkNSP",
                  "rules": Array [
                    "
  display: flex;
  max-width: 360px;
  flex: 1;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-ifAKCX fbkNSP"
            >
              <Styled(Styled(Component))
                isLeft={true}
                onClick={[Function]}
                selected={false}
                testId="add-fixed"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-gzVnrw",
                        "isStatic": false,
                        "lastClassName": "jVyLDY",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  width: 47%
  padding: 28px 0;
  margin-right: ",
                          [Function],
                          ";
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-gzVnrw",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isLeft={true}
                  onClick={[Function]}
                  selected={false}
                  testId="add-fixed"
                >
                  <Component
                    className="sc-bwzfXH sc-gzVnrw iLZNNl"
                    isLeft={true}
                    onClick={[Function]}
                    selected={false}
                    testId="add-fixed"
                  >
                    <button
                      className="sc-bwzfXH sc-gzVnrw iLZNNl"
                      data-testid="add-fixed"
                      onClick={[Function]}
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-EHOje",
                                "isStatic": false,
                                "lastClassName": "fvXRgE",
                                "rules": Array [
                                  "
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-EHOje",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-EHOje fvXRgE"
                          >
                            ONBOARDING_TYPE_FIXED_STANDARD_OFFICE
                          </div>
                        </StyledComponent>
                      </styled.div>
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bZQynM",
                                "isStatic": false,
                                "lastClassName": "jBQPio",
                                "rules": Array [
                                  "
  color: #828286;
  font-size: 14px;
  font-weight: 400;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bZQynM",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bZQynM jBQPio"
                          >
                            ONBOARDING_TYPE_FIXED_SUBTITLE
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
              <Styled(Styled(Component))
                onClick={[Function]}
                selected={false}
                testId="add-flex"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-gzVnrw",
                        "isStatic": false,
                        "lastClassName": "jVyLDY",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  width: 47%
  padding: 28px 0;
  margin-right: ",
                          [Function],
                          ";
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-gzVnrw",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  selected={false}
                  testId="add-flex"
                >
                  <Component
                    className="sc-bwzfXH sc-gzVnrw jVyLDY"
                    onClick={[Function]}
                    selected={false}
                    testId="add-flex"
                  >
                    <button
                      className="sc-bwzfXH sc-gzVnrw jVyLDY"
                      data-testid="add-flex"
                      onClick={[Function]}
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-EHOje",
                                "isStatic": false,
                                "lastClassName": "fvXRgE",
                                "rules": Array [
                                  "
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-EHOje",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-EHOje fvXRgE"
                          >
                            ONBOARDING_TYPE_FLEX_COWORKING
                          </div>
                        </StyledComponent>
                      </styled.div>
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bZQynM",
                                "isStatic": false,
                                "lastClassName": "jBQPio",
                                "rules": Array [
                                  "
  color: #828286;
  font-size: 14px;
  font-weight: 400;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bZQynM",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bZQynM jBQPio"
                          >
                            ONBOARDING_TYPE_FLEX_SUBTITLE
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Type>
`;
